mbed_devices/_internal/windows/windows_identifier.py

```diff
@@ -0,0 +1,68 @@
+"""Defines a Windows identifier."""
+from typing import NamedTuple, Any, Optional
+
+from mbed_devices._internal.windows.component_descriptor_utils import is_undefined_value
+
+INSTANCE_ID_CHARACTER = "&"
+
+
+class WindowsUID(NamedTuple):
+    """Definition of a Windows Universal ID."""
+
+    UID: str  # Processed string which may be the closest to the Universal ID.
+    raw_uid: str  # The String from which the UID was derived. Raw form as it can be a
+    # combination of other elements e.g. 8&3432E179&0&000
+    serial_number: Optional[str]  # a serial number
+
+    def __eq__(self, other: Any) -> bool:
+        """Defines the equality checker."""
+        if not other or not isinstance(other, WindowsUID):
+            return False
+        if (
+            self.UID == other.UID
```

Comment:
If `UID` is always a subset of `raw_uid` there's probably not much point adding it to the equality check as we're already checking `raw_uid`.

Reply:
UID is not always a subset. it can be a lowercase subset.

Reply:
e3b58499695e9796bf1b0ed63d97dd330c1692a8